pkg/cmd/gist/edit/edit.go

```diff
@@ -306,36 +316,46 @@ func editRun(opts *EditOptions) error {
 		return nil
 	}
 
-	return updateGist(apiClient, host, gist)
+	return updateGist(apiClient, host, gistToUpdate)
 }
 
-func updateGist(apiClient *api.Client, hostname string, gist *shared.Gist) error {
-	body := shared.Gist{
-		Description: gist.Description,
-		Files:       gist.Files,
-	}
+// https://docs.github.com/en/rest/gists/gists?apiVersion=2022-11-28#update-a-gist
+type gistToUpdate struct {
+	// The id of the gist to update. Does not get marshalled to JSON.
+	id string
+	// The description of the gist
+	Description string `json:"description"`
+	// The gist files to be updated, renamed or deleted. The key must match the current
+	// filename of the file to change. To delete a file, set the value to nil
+	Files map[string]*gistFileToUpdate `json:"files"`
+}
 
-	path := "gists/" + gist.ID
+type gistFileToUpdate struct {
+	// The new content of the file
+	Content string `json:"content"`
+	// The new name for the file
+	Filename string `json:"filename,omitempty"`
```

Comment:
I wouldn't mind calling this `NewFilename` for clarities sake. The relationship between the map key being the filename to change and this as `Filename` is not obvious. The JSON struct tag would remain the same.

Reply:
That makes sense to me.